src/fc/fc_domain.c

```diff
@@ -386,7 +390,7 @@ fcConfigAndStartVM(virFCDriver *driver G_GNUC_UNUSED,
     }
 
     if (virFCMonitorSetDisk(vmPrivateData->socketpath, "rootfs",
-                            root_device->src->path, true, false) < 0) {
+                            root_device->src->path, true, root_device->src->readonly) < 0) {
```

Comment:
if you add the tag `<readonly/>` under the `<disk>` device definition, the rootfs will be in read-only mode